ocaml/xapi/import_raw_vdi.ml

```diff
@@ -111,20 +112,27 @@ let import vdi (req: Request.t) (s: Unix.file_descr) _ =
             | Some sr ->
               debug "Checking whether localhost can see SR: %s" (Ref.string_of sr);
               if (Importexport.check_sr_availability ~__context sr)
-              then localhost_handler rpc session_id vdi req s
-              else
+              then localhost_handler rpc session_id vdi req s  
+              else begin
+              try   
                 let host = Importexport.find_host_for_sr ~__context sr in
                 let address = Db.Host.get_address ~__context ~self:host in
                 return_302_redirect req s address;
                 None
+              with e ->
+                error "Caught exception in import handler: %s" (ExnHelper.string_of_exn e);
+                ignore(Xapi_http.task_id_of_req req |> Stdext.Opt.map (fun task_id -> TaskHelper.failed ~__context:(Context.from_forwarded_task task_id) e));
+                Http_svr.headers s (Http.http_400_badrequest ());
+                None
+              end;
             | None ->
               error "Require an SR or VDI to import";
-              TaskHelper.failed ~__context Api_errors.(Server_error(vdi_missing,[]));
+              let e = Api_errors.(Server_error(vdi_missing,[])) in 
+              ignore(Xapi_http.task_id_of_req req |> Stdext.Opt.map (fun task_id -> TaskHelper.failed ~__context:(Context.from_forwarded_task task_id) e));
```

Comment:
Since we are using this expression at two places, we could extract it into a local `fail_task_in_request e` helper function inside this `import` function (where `e` is our exception).